packages/eslint-plugin/tests/rules/no-unnecessary-type-assertion.test.ts

```diff
@@ -16,6 +16,27 @@ const ruleTester = new RuleTester({
 ruleTester.run('no-unnecessary-type-assertion', rule, {
   valid: [
     `
+      let x = 1; // typeof x = number
+      let y = 1 as 1; // typeof y = 1
+      const x = 1; // typeof x = 1
+      let y = x; // typeof y = number
+      let z2 = x as 1; // okay - typeof z2 = 1
+      const foo = 3 as number;
+      const foon = <number>3;
+      const foom = 3 as number;
```

Comment:
[Testing] We generally prefer test cases to be small and self-contained - rather than having multiple errors / types of parameter properties in one case (https://typescript-eslint.io/contributing/local-development#granular-unit-tests)

Could you please split them up?